src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
```

Comment:
There is a typo in the constant name 'DEAFULT_SIZE'. It should be 'DEFAULT_SIZE'. This is an issue related to the task requirements as it can lead to confusion and is not following standard naming conventions.